src/openjd/model/v2023_09/_model.py

```diff
@@ -2393,6 +2432,42 @@ class JobTemplate(OpenJDModel_v2023_09):
         rename_fields={"parameterDefinitions": "parameters"},
     )
 
+    @field_validator("extensions")
+    @classmethod
+    def _unique_extension_names(
+        cls, value: Optional[ExtensionNameList]
+    ) -> Optional[ExtensionNameList]:
+        if value is not None:
+            return validate_unique_elements(
+                value, item_value=lambda v: v, property="extension name"
+            )
+        return value
+
+    @field_validator("extensions")
+    @classmethod
+    def _permitted_extension_names(
+        cls, value: Optional[ExtensionNameList], info: ValidationInfo
+    ) -> Optional[ExtensionNameList]:
+        context = cast(ModelParsingContext, info.context)
+        if value is not None:
+            # Before processing the extensions field, context.extensions is the list of supported extensions.
+            # Take the intersection of the input supported extensions with what is implemented
+            # in this list, as the implementation needs to support an extension for it to be supported.
+            supported_extensions = context.extensions.intersection(cls.supported_extension_names())
+
+            unsupported_extensions = set(value).difference(supported_extensions)
+            if unsupported_extensions:
+                raise ValueError(
+                    f"Unsupported extension names: {', '.join(sorted(unsupported_extensions))}"
+                )
+
+            # After processing the extensions field, context.extensions is the list of
+            # extension names used by the template.
+            context.extensions = set(value)
+        else:
+            context.extensions = set()
+        return value
```

Comment:
It looks funny that we're mutating the input parameter and returning a value. Is this a Pydantic pattern?

Also should `context.extensions` be a list instead of a set?